src/Converter/Feature2Texture.js

```diff
@@ -34,94 +34,86 @@ function _drawPolygon(ctx, vertices, indices, properties = {}, style, size, exte
     }
 
     // draw line polygon
-    if (style.stroke || properties.stroke) {
-        strokeStyle(style, properties, ctx, invCtxScale);
+    if (style.stroke.color) {
+        strokeStyle(style, ctx, invCtxScale);
         ctx.stroke();
     }
 
     // fill polygon
-    if (style.fill || properties.fill) {
-        fillStyle(style, properties, ctx);
+    if (style.fill.color) {
+        fillStyle(style, ctx);
         ctx.fill();
     }
 }
 
-function fillStyle(style, properties, ctx) {
-    const fill = style.fill || properties.fill;
-    if (ctx.fillStyle !== fill) {
-        ctx.fillStyle = fill;
+function fillStyle(style, ctx) {
+    if (ctx.fillStyle !== style.fill.color) {
+        ctx.fillStyle = style.fill.color;
     }
-    const alpha = style.fillOpacity || properties['fill-opacity'] || 1.0;
-    if (alpha !== ctx.globalAlpha && typeof alpha == 'number') {
-        ctx.globalAlpha = alpha;
+    if (style.fill.opacity !== ctx.globalAlpha) {
+        ctx.globalAlpha = style.fill.opacity;
     }
 }
 
-function strokeStyle(style, properties, ctx, invCtxScale) {
-    const stroke = style.stroke || properties.stroke;
-    if (ctx.strokeStyle !== stroke) {
-        ctx.strokeStyle = stroke;
+function strokeStyle(style, ctx, invCtxScale) {
+    if (ctx.strokeStyle !== style.stroke.color) {
+        ctx.strokeStyle = style.stroke.color;
     }
-    const width = Math.round((style.strokeWidth || properties['stroke-width'] || 2.0)) * invCtxScale;
+    const width = Math.round((style.stroke.width || 2.0)) * invCtxScale;
     if (ctx.lineWidth !== width) {
         ctx.lineWidth = width;
     }
-    const alpha = style.strokeOpacity || properties['stroke-opacity'] || 1.0;
+    const alpha = style.stroke.opacity || 1.0;
     if (alpha !== ctx.globalAlpha && typeof alpha == 'number') {
         ctx.globalAlpha = alpha;
     }
 }
 
 function drawPoint(ctx, x, y, style = {}, invCtxScale) {
     ctx.beginPath();
-    const opacity = style.pointOpacity || 1.0;
+    const opacity = style.point.opacity || 1.0;
```

Comment:
Let's say for some reason I have a full transparent point: this condition is not working for me here. This needs to move to something like `style.point.opacity == undefined ? 1.0 : style.point.opacity`. Or maybe you could do something about it directly in `Style`, I dunno.

Reply:
yes, thanks